CDTDatastore/HTTP/CDTHTTPInterceptorContext.h

```diff
@@ -39,7 +48,8 @@ NS_ASSUME_NONNULL_BEGIN
  *
  *  @param request the request this context should represent
  **/
-- (instancetype)initWithRequest:(NSMutableURLRequest *)request NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithRequest:(NSMutableURLRequest *)request
+                          state:(NSMutableDictionary *)state NS_DESIGNATED_INITIALIZER;
```

Comment:
It's an interesting point. If they can return a new context, then they can destroy the state for other interceptors. Of course they can modify the state as much as they want inside the interceptor anyway.